Magento2/Sniffs/Legacy/WidgetXMLSniff.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento2\Sniffs\Legacy;
+
+use DOMDocument;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+/**
+ * Test for obsolete nodes/attributes in the widget.xml
+ */
+class WidgetXMLSniff implements Sniff
+{
+    private const ERROR_CODE_OBSOLETE = 'FoundObsoleteNode';
+    private const ERROR_CODE_FACTORY = 'FoundFactory';
+    private const ERROR_CODE_XML = 'WrongXML';
+
+    /**
+     * @inheritdoc
+     */
+    public function register(): array
+    {
+        return [
+            T_INLINE_HTML
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        if ($stackPtr > 0) {
+            return;
+        }
+
+        try {
+            $xml = simplexml_load_string($this->getFormattedXML($phpcsFile));
```

Comment:
Are you sure on this? `simplexml_load_string` will return `false` on failure, the only method that could throw an exception is `getTokensAsString()`, but shouldn't when passing `0` as start value.

Reply:
@svera I added both cases.